lib/utils.js

```diff
@@ -464,7 +464,7 @@ const decoders = {
     if (typeof data === 'string')
       data = Buffer.from(data, 'latin1');
     try {
-      const decoder = new TextDecoder(this);
+      const decoder = new TextDecoder();
```

Comment:
This isn't the correct change. Instead,

```js
other: (data, hint) => {
```

should be:

```js
other: function(data, hint) {
```

Reply:
Hmm, maybe we can make that a bit clearer? Something like:

```js
other: (charset) => (data, hint) =>
```

Reply:
That isn't correct either. `getDecoder()` returns functions, not values. That's why `bind()` was being used. All that needs to be done is changing from an arrow function to a non-arrow function so it can be correctly bound and that `this` will correctly refer to the bound charset name.

Reply:
I think using `bind()` will lead to better efficiency as V8 should be able to optimize better rather than returning a completely "different" function every time.

Reply:
Yes, it requires a change to the call site (in getDecoder) as well (see the latest commit). If you feel it has other issues I can revert to using bind.

Reply:
Because I am curious: https://jsbench.me/1ilnv78d75/1

Looks like .bind has some overhead and slightly loses out, at least on my machine haha. V8 must do some clever things to optimize a repetitive function creation.

<img width="801" alt="image" src="https://github.com/mscdex/busboy/assets/40680/5cb21cf2-5b58-46b1-96ef-e7468b1e99a6">